command/agent/consul/int_test.go

```diff
@@ -131,7 +141,7 @@ func TestConsul_Integration(t *testing.T) {
 		serviceClient.Run()
 		close(consulRan)
 	}()
-	tr := client.NewTaskRunner(logger, conf, logUpdate, taskDir, alloc, task, vclient, serviceClient)
+	tr := client.NewTaskRunner(logger, conf, db, logUpdate, taskDir, alloc, task, vclient, serviceClient)
```

Comment:
This makes me wish our use of boltdb was wrapped in a typed state manager we could put behind an interface and mock for tests. There's no reason for these tests to actually exercise any of the state code.

It's fine it's that out of scope as I understand it'd require refactoring basically every change in this PR.